Nucleo_l4r5zi: Enable LPUART_1 as console

erwango opened the pull request:
LPUART_1 is default serial port used on ST-Link VPC for board nucleo_l4r5zi.
It is configured to use Port G pins which require dedicated power supply.

This PR does the needful for port G power activation, LPUART_1 configuration on board and some clean up in STM32 UART Kconfig symbols.

Fixes #12092

erwango committed:
drivers/gpio: stm32l4: PortG[15:2] requires dedicated VDDio2

On STM32L4, 14 pins (Port G[15:2]), require external power supply.
Activate this at port G init when PWR_CR2_IOSV is defined.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/serial: stm32: rename symbol UART_LPUART_1 to LPUART_1

Following recent renaming of STM32 UART Kconfig UART symbols,
LPUART was named as UART_LPUART_1.
Rename to LPUART_1.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/serial: stm32: Remove superfluous dependency in Kconfig

In Kconfig.stm32, all UART port symbols were defined with a dependency
on symbol UART_STM32. This is redundant since they are located under
if UART_STM32 condition.


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@psidhu, please review, as you were the one pushing this board
@gon1332, you might reuse this for nucleo_l496zg
@cybertale, please take it into account in #12024

cybertale commented:
@erwango OK, I'll see what I can do.

gon1332 commented on drivers/gpio/gpio_stm32.c:
```diff
@@ -197,6 +197,20 @@ static int gpio_stm32_init(struct device *device)
 
+#ifdef PWR_CR2_IOSV
+	if (cfg->port == STM32_PORTG) {
+		/* Port G[15:2] tequires external power supply */
```
typo on requires

erwango replied:
ack

psidhu commented on boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.dts:
```diff
@@ -70,7 +70,7 @@ arduino_serial: &usart3 {};
 
-&usart3 {
+&lpuart1 {
 	current-speed = <115200>;
```
You should add another node, note replace usart3. usart3 is used I think for the Arduino shield.

erwango replied:
Right, this is actually a hack that went in. Fixed

erwango committed:
boards: nucleo_l4r5zi: set lpuart1 as VPC port

lpuart1 is the default port used by ST-Link VPC.
Update board description to match out of box board configuration.

Fixes #12092

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

gon1332 commented:
With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.

I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.

gon1332 commented:
> With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.
> 
> I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.

Oh wait! You haven't used USB ACM. I will recheck.

gon1332 commented:
> > With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.
> > I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.
> 
> Oh wait! You haven't used USB ACM. I will recheck.

I didn't work. The output is empty.
The strange thing is that when I choose LPUART_1 from DTS for console output, the application with which I flash the board is not running.

erwango commented:
> The strange thing is that when I choose LPUART_1 from DTS for console output, the application with which I flash the board is not running.

I've tested these changes on top of you branch and worked. Did you update Kconfig.defconfig?

gon1332 commented:
@erwango I've checked your fix, with success at this time. How do you see it merged in my PR (https://github.com/zephyrproject-rtos/zephyr/pull/11265)?

Which one will get merged first? Do you want me to have the necessary changes ready?

erwango commented:
@gon1332 , let try to merge this one first. Can you approve it if finally validated?
@psidhu, txs for the review, I fixed your point, can you review again?

psidhu reviewed (approved):
LGTM

gon1332 commented:
@erwango Nice! 2 out of 2.
Is it time yet to merge this PR?

erwango commented:
@galak, this one is ready for merge if you have some time
